ui/src/app/components/app-router.tsx

```diff
@@ -69,17 +69,14 @@ export const AppRouter = ({children}): React.ReactElement => <BrowserRouter>{chi
 
 export const RouteLink = ({path, style = {}, children}): React.ReactElement => <Link style={{...style}} to={path}>{children}</Link>;
 
-export const AppRoute = ({path, data = {}, guards = [], component: Component, exact = true}): React.ReactElement => {
-  const routeParams = useParams();
-  const routeHistory = useHistory();
+export const AppRoute = ({path, guards = [], exact= true, children}): React.ReactElement => {
+  const { redirectPath = null } = fp.find(({allowed}) => !allowed(), guards) || {};
 
-  return <Route exact={exact} path={path} render={
-    () => {
-      const { redirectPath = null } = fp.find(({allowed}) => !allowed(), guards) || {};
-      return redirectPath
+  return <Route exact={exact} path={path}>
+    {redirectPath
```

Comment:
not part of your changes but I wonder if this works correctly if the `redirectPath` is an empty string. presumably, the user intent would be to redirect to root but it would flow into the falsy path ATM.

Reply:
Ooh good point. Yes, we should do something about that.